web/pdf_page_view.js

```diff
@@ -320,6 +331,7 @@ var PDFPageView = (function PDFPageViewClosure() {
     draw: function PDFPageView_draw() {
       if (this.renderingState !== RenderingStates.INITIAL) {
         console.error('Must be in new state before drawing');
+        return Promise.resolve(undefined);
```

Comment:
> Not sure if we want to change both. For pages, we fall through to start new rendering (e.g. for different scale). For thumbs, we don't care about scale or if image is set after page rendering.

Really good point about the differences between pages/thumbnails! After thinking about this, I'm no longer sure that we want to make too many functionaly changes here.

The only change I've made in the updated patch, is to add a `this.reset();` call for the `PDFPageView` case, since I think that we should ensure that we reset various state on fall-through.
Please do let me know if you don't think this is appropriate!